keyboards/phage_studio/pila87/keymaps/default/keymap.c

```diff
@@ -0,0 +1,200 @@
+       │      │   │   │   │   │   │   │   │   │   │   │   │        │
+       ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────────┤     ┌───┐
+       │        │   │   │   │   │   │   │   │   │   │   │          │     │   │
+       ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤ ┌───┼───┼───┐
+       │    │    │    │                        │    │    │    │    │ │   │   │   │
+       └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘ └───┴───┴───┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14       15       16     */
+    [_FN3]  = LAYOUT_tkl_ansi(
+                _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,          _______,
+                _______, _______, _______,                   _______,                                     _______, _______, _______, _______, _______, _______, _______
+            ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                        rgb_matrix_enable_noeeprom();
+                    }
+                    break;
+                }
+            }
+            return false;
+	}
+    return true;
+}
+
+bool rgb_matrix_indicators_user(void) {
+    HSV      hsv = rgb_matrix_config.hsv;
+    uint8_t time = scale16by8(g_rgb_timer, qadd8(32, 1));
+    hsv.h        = time;
+    RGB      rgb = hsv_to_rgb(hsv);
+
+    if ((rgb_matrix_get_flags() & LED_FLAG_ALL)) {
+        if (host_keyboard_led_state().caps_lock) {
+            rgb_matrix_set_color(40, rgb.r, rgb.g, rgb.b);
+        }
+    } else {
+        if (host_keyboard_led_state().caps_lock) {
+            rgb_matrix_set_color(40, rgb.r, rgb.g, rgb.b);
+        } else {
+            rgb_matrix_set_color(40, 0, 0, 0);
+        }
+    }
```

Comment:
This is a bit more compact. ;)
```suggestion
    if (host_keyboard_led_state().caps_lock) {
        rgb_matrix_set_color(40, rgb.r, rgb.g, rgb.b);
    } else if (!(rgb_matrix_get_flags() & LED_FLAG_INDICATOR)) {
        rgb_matrix_set_color(40, 0, 0, 0);
    }
```

Also, it may be worth moving all of this to pila87.c, and using the kb versions of the functions.

Reply:
I've just resolved this one, thanks